pkg/console/subresource/consoleserver/config_builder.go

```diff
@@ -317,32 +317,74 @@ func (b *ConsoleServerCLIConfigBuilder) customization() Customization {
+			for categoryIndex, category := range b.devCatalogCustomization.Categories {
+				var subcategories []DeveloperConsoleCatalogCategoryMeta = nil
+				if category.Subcategories != nil {
+					subcategories = make([]DeveloperConsoleCatalogCategoryMeta, len(category.Subcategories))
+					for subcategoryIndex, subcategory := range category.Subcategories {
+						subcategories[subcategoryIndex] = mapMeta(subcategory)
+					}
+				}
+				categories[categoryIndex] = DeveloperConsoleCatalogCategory{
+					DeveloperConsoleCatalogCategoryMeta: mapMeta(category.DeveloperConsoleCatalogCategoryMeta),
+					Subcategories:                       subcategories,
+				}
 			}
 		}
+		if !reflect.DeepEqual(b.devCatalogCustomization.Types, operatorv1.DeveloperConsoleCatalogTypesState{}) {
 
-		categories := make([]DeveloperConsoleCatalogCategory, len(b.devCatalogCustomization.Categories))
-		for categoryIndex, category := range b.devCatalogCustomization.Categories {
-			var subcategories []DeveloperConsoleCatalogCategoryMeta = nil
-			if category.Subcategories != nil {
-				subcategories = make([]DeveloperConsoleCatalogCategoryMeta, len(category.Subcategories))
-				for subcategoryIndex, subcategory := range category.Subcategories {
-					subcategories[subcategoryIndex] = mapMeta(subcategory)
+			var disabledTypes []string = nil
+			var enabledTypes []string = nil
+
+			if b.devCatalogCustomization.Types.State == operatorv1.CatalogTypesState(CatalogTypeDisabled) && b.devCatalogCustomization.Types.Disabled != nil {
+				disabledTypes = *b.devCatalogCustomization.Types.Disabled
+			}
+
+			if b.devCatalogCustomization.Types.State == operatorv1.CatalogTypesState(CatalogTypeEnabled) && b.devCatalogCustomization.Types.Enabled != nil {
+				enabledTypes = *b.devCatalogCustomization.Types.Enabled
+			}
+
+			if b.devCatalogCustomization.Types.State == operatorv1.CatalogTypesState(CatalogTypeDisabled) {
+				catalogTypes = DeveloperConsoleCatalogTypesState{
+					State:    CatalogTypeDisabled,
+					Disabled: &disabledTypes,
 				}
 			}
-			categories[categoryIndex] = DeveloperConsoleCatalogCategory{
-				DeveloperConsoleCatalogCategoryMeta: mapMeta(category.DeveloperConsoleCatalogCategoryMeta),
-				Subcategories:                       subcategories,
+
+			if b.devCatalogCustomization.Types.State == operatorv1.CatalogTypesState(CatalogTypeEnabled) {
+				catalogTypes = DeveloperConsoleCatalogTypesState{
+					State:   CatalogTypeEnabled,
+					Enabled: &enabledTypes,
+				}
 			}
 		}
 
-		conf.DeveloperCatalog = &DeveloperConsoleCatalogCustomization{
-			Categories: &categories,
+		if len(categories) > 0 {
+			conf.DeveloperCatalog = &DeveloperConsoleCatalogCustomization{
+				Categories: &categories,
+				Types:      catalogTypes,
+			}
+		} else {
+			conf.DeveloperCatalog = &DeveloperConsoleCatalogCustomization{
+				Categories: &[]DeveloperConsoleCatalogCategory{},
+				Types:      catalogTypes,
+			}
 		}
```

Comment:
Looks same to me, categories is an empty error, or you create one. Isn't this enough here?

```suggestion
	conf.DeveloperCatalog = &DeveloperConsoleCatalogCustomization{
		Categories: &categories,
		Types:      catalogTypes,
	}
```